wallet/src/account.rs

```diff
@@ -20,7 +22,18 @@ pub struct Wallet {
 #[derive(Debug)]
 pub struct State {
     value: Value,
-    counter: SpendingCounter,
+    counters: Vec<SpendingCounter>,
```

Comment:
Is there a reason you're not using `SpendingCounterIncreasing`?

Reply:
I was using it initially, but I was a bit unsure of a few minor things and replaced it with this, as the logic is really simple anyway. Mostly it seemed a bit coupled with the ledger.

For example, I was considering on relaxing the property of the constructor taking the spending counters sorted, as they have the lane inside anyway (I didn't do it, but I think could make sense). I still find strange that the argument is called `set` but still requires the outputs to be sorted. Also, `get_valid_counters` doesn't make any promises about the order.

Also, `next_verify` being mutable and returning a `LedgerError` is a bit strange. I know I can just ignore the error type and replace it with the error type in here, but... I don't know. And I guess cloning and then mutating is fine, but it seems more natural to just build the new vec.

I don't really have a strong posture on this though, I just wasn't sure about creating unnecessary coupling, if you think it's better to use that then it's fine for me.

Reply:
I just have basically the same problem in testing wallets. On one side it's nice to have things tied to the ledger (e.g, you don't accidentally use more lanes that are available), but as you noted there are a few downsides. 
However, IIRC also `sub` operations on wallet value return a `LedgerError`

Reply:
I decided to use it anyway to keep things consistent, the issues that I had were really minor anyway. If we need more flexibility in the future we can just change it.

Reply:
Decided to use it anyway, the issues that I had were really minor (and I actually had misunderstood part of the api), we can just change it in the future if we need more flexibility or something.